src/routes/auto-connect.ts

```diff
@@ -37,6 +37,11 @@ export const autoConnectWallet = async (options: {
       syncing: false
     }
 
+    // Only one wallet can be connected in larp mode
+    if (larpMode$.value && wallets$.value.length) {
+      wallets$.value.forEach(async wallet => await db.wallets.delete(wallet.id))
+    }
```

Comment:
```suggestion
    if (larpMode$.value && wallets$.value.length) {
      for (const wallet of wallets$.value) {
         await db.wallets.delete(wallet.id) 
      }
    }
```
A for each loop will not wait for all of the wallets to be deleted before moving on to the next expression, so could lead to some weird race conditions. So you could use a for of loop or use `Promise.all(wallets$.value.map(wallet => db.wallets.delete(wallet.id))` to ensure all wallets are deleted before adding the new one.